File: meldingsvarsler-fasit-forslag/src/test/java/no/sonat/meldingsvarsler/MeldingRepositoryStub.java
package no.sonat.meldingsvarsler;

import no.sonat.meldingsvarsler.epost.EpostMelding;
import no.sonat.meldingsvarsler.facebook.FacebookMelding;
import no.sonat.meldingsvarsler.meldinger.MeldingRepository;
import no.sonat.meldingsvarsler.meldinger.MeldingRepositoryWriter;
import no.sonat.meldingsvarsler.sms.SMSMelding;

import java.util.ArrayList;
import java.util.List;

public class MeldingRepositoryStub implements MeldingRepository, MeldingRepositoryWriter {
    List<Melding> meldinger = new ArrayList<>();
    @Override
    public List<Melding> hentMeldinger(){
        meldinger.add(new SMSMelding("Vårens kleskolleksjon har kommet. Løp og kjøp. Førstemann til mølla!!"));
        meldinger.add(new EpostMelding("Løp og kjøp. Førstemann til mølla!!",
                "Vårens kleskolleksjon har kommet. "));
        meldinger.add(new FacebookMelding("Vårens kleskolleksjon har kommet. Løp og kjøp. Førstemann til mølla!! Del på Facebook."));
        return meldinger;
    }

    @Override
    public void leggTilMelding(Melding melding) {

    }
}

File: meldingsvarsler-fasit-forslag/Tests/MeldingRepositoryStub.cs
using System.Collections.Generic;
using App;

public class MeldingRepositoryStub : IMeldingRepository,
                                     IMeldingRepositoryWriter
{
    readonly List<IMelding> meldinger = new List<IMelding>();

    public List<IMelding> HentMeldinger()
    {
        meldinger.Add(new SmsMelding("Vårens kleskolleksjon har kommet. Løp og kjøp. Førstemann til mølla!!"));
        meldinger.Add(new EpostMelding("Løp og kjøp. Førstemann til mølla!!",
                                       "Vårens kleskolleksjon har kommet. "));
        meldinger.Add(
            new FacebookMelding(
                "Vårens kleskolleksjon har kommet. Løp og kjøp. Førstemann til mølla!! Del på Facebook."));
        return meldinger;
    }

    public void LeggTilMelding(IMelding melding) { }
}